File: neetcode/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {val:i for i, val in enumerate(nums)}
        for i in range(len(nums)):
            remainder = target - nums[i]
            if remainder in lookup.keys() and lookup[remainder] != i:
                return [i, lookup[remainder]]
        
# @lc code=end

File: neetcode/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	hm := make(map[int]int)
	for i, num := range nums {
		remainder := target - num
		if remainder_index, ok := hm[remainder]; ok {
			return []int{i, remainder_index}
		}
		hm[num] = i
	}
	return []int{}
}

// @lc code=end

